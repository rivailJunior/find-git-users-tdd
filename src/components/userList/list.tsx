import { iUser } from "../../interfaceModel/user";
import TableList from '../table'

type iList = {
    loading: boolean;
    users: iUser[]
}

export default function List({ loading, users }: iList) {
    return (
        <div>
            <h3>Lista de Usu√°rios</h3>
            {loading ? (
                <div>Loading...</div>
            ) : (

                <TableList
                    data={users}
                    headerProperties={[{ label: 'Login' }]}
                    bodyProperties={[{ key: 'login' }]}
                />
            )}
        </div>
    );
}
