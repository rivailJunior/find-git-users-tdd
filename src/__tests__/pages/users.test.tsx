import { render, screen } from "@testing-library/react"
import Users from '../../pages/users';
import RequestApi from '../../helper/apiHelper';
import { userList } from "../../mock/userList";
import { iUser } from "../../interfaceModel/user";

let mockUsers: iUser[] = [];
beforeAll(() => {
    mockUsers = userList;
    return mockUsers
})

jest.mock('../../helper/apiHelper');
describe('When Users page is called', () => {
    describe('and everything is ok', () => {
        it('Snapshot with users', async () => {
            (RequestApi.getUsers as jest.Mock).mockResolvedValue(mockUsers);
            const { container } = render(<Users />);
            await screen.findByTestId('userList');
            expect(container.firstChild).toMatchSnapshot()
        })
        it('Show User List', async () => {
            (RequestApi.getUsers as jest.Mock).mockResolvedValue(mockUsers);
            render(<Users />);
            const userList = await screen.findByTestId('userList');
            expect(userList.children.length).toBe(3);
        });

    });

    describe('or when there is no user on list', () => {
        it('Show Empty List', async () => {
            (RequestApi.getUsers as jest.Mock).mockResolvedValue([]);
            render(<Users />);
            const loading = await screen.findByText(/Loading.../);
            expect(loading).toBeInTheDocument()
        })
    })


})