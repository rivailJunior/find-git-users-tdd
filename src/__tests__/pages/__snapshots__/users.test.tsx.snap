// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When Users page is called and everything is ok Snapshot with users 1`] = `
<div>
  <div>
    <h3>
      Lista de Usu√°rios
    </h3>
    <table
      class="table table-striped table-bordered table-hover"
    >
      <thead
        data-testid="tableHeader"
      >
        <tr>
          <th>
            Login
          </th>
        </tr>
      </thead>
      <tbody
        data-testid="userList"
      >
        <tr>
          <td>
            mojombo
          </td>
        </tr>
        <tr>
          <td>
            defunkt
          </td>
        </tr>
        <tr>
          <td>
            pjhyett
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
